tests/unit/style/properties/serialization.rs

```diff
@@ -1114,4 +1114,31 @@ mod shorthand_serialization {
             assert_eq!(s, "none");
         }
     }
+
+    mod effects {
+        pub use super::*;
+        pub use style::properties::longhands::box_shadow::SpecifiedValue as BoxShadow;
+        pub use style::values::specified::Shadow;
+        #[test]
+        fn box_shadow_should_serialize_correctly() {
+            let mut properties = Vec::new();
+            let color = Some(CSSColor {
+                        parsed: ComputedColor::RGBA(RGBA { red: 1f32, green: 0f32, blue: 0f32, alpha: 1f32 }),
+                        authored: None
+                        });
+
+
+            let shadow_val = Shadow { offset_x: Length::from_px(1f32), offset_y: Length::from_px(2f32),
+            blur_radius: Length::from_px(3f32), spread_radius: Length::from_px(4f32), color: color, inset: false };
+            let shadow_decl = DeclaredValue::Value(BoxShadow(vec![shadow_val]));
+            properties.push(PropertyDeclaration:: BoxShadow(shadow_decl));
+
+
+             let serialization = shorthand_properties_to_string(properties);
+             assert_eq!(serialization, "box-shadow: 1px 2px 3px 4px rgb(255, 0, 0);");
+        }
+    }
+
+
+    }
```

Comment:
Is there a way to get to know what is causing the build to fail. I removed that bracket it still fails. I don't want to keep bugging you so was wondering if I can fix the build errors.